src/cluster.c

```diff
@@ -2019,6 +2158,9 @@ void clusterProcessPingExtensions(clusterMsg *hdr, clusterLink *link) {
         if (type == CLUSTERMSG_EXT_TYPE_HOSTNAME) {
             clusterMsgPingExtHostname *hostname_ext = (clusterMsgPingExtHostname *) &(ext->ext[0].hostname);
             ext_hostname = hostname_ext->hostname;
+        } else if (type == CLUSTERMSG_EXT_TYPE_SHARDID) {
+            clusterMsgPingExtShardId *shardid_ext = (clusterMsgPingExtShardId *) &(ext->ext[0].shard_id);
+            memcpy(ext_shardid, shardid_ext->shard_id, CLUSTER_NAMELEN);
```

Comment:
We can just extract the pointer and pass it to updateShardId(). Doing a copy here is just extra work since we're going to copy it again later anyways.

Reply:
+1